File: Dike/dike.py
import sys

fi = open("input.txt")
fo = open("output.txt", "w")
sys.stdin = fi
sys.stdout = fo

n = int(input())
lf = 0
rt = min(n + 1, 2_000_000)

while rt - lf > 1:
    m = (rt + lf)//2
    s = m*(m + 1)*(m + 5)//6 - 1
    if s > n:
        rt = m
    else:
        lf = m

print(lf)

fo.close()
fi.close()

File: Dike/dike.cpp
#include <bits/stdc++.h>

using namespace std;

const int64_t M = 2'000'000;

int main()
{
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int64_t n, m, sum;
    int lf, rt;
    cin >> n;
    lf = 0, rt = min(n + 1, M);

    while (rt - lf > 1)
    {
        m = (lf + rt) / 2;
        sum = m * (m + 1) * (m + 5) / 6 - 1;
        if (sum > n)
            rt = m;
        else
            lf = m;
    }

    cout << lf << endl;

    return 0;
}